x-pack/legacy/plugins/lens/public/indexpattern_plugin/layerpanel.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React, { useState, useMemo } from 'react';
+import {
+  EuiComboBox,
+  // @ts-ignore
+  EuiHighlight,
+  EuiButtonEmpty,
+  EuiIcon,
+  EuiIconTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { I18nProvider } from '@kbn/i18n/react';
+import { DatasourceLayerPanelProps } from '../types';
+import { IndexPatternPrivateState } from './indexpattern';
+import { isLayerTransferable, updateLayerIndexPattern } from './state_helpers';
+
+export interface IndexPatternLayerPanelProps extends DatasourceLayerPanelProps {
+  state: IndexPatternPrivateState;
+  setState: (newState: IndexPatternPrivateState) => void;
+}
+
+export function LayerPanel({ state, setState, layerId }: IndexPatternLayerPanelProps) {
+  const [isChooserOpen, setChooserOpen] = useState(false);
+  const indexPatterns = useMemo(
+    () =>
+      Object.values(state.indexPatterns)
+        .filter(indexPattern => indexPattern.id !== state.layers[layerId].indexPatternId)
+        .map(indexPattern => ({
+          ...indexPattern,
+          isTransferable: isLayerTransferable(state.layers[layerId], indexPattern),
+        })),
+    [state.indexPatterns, layerId, state.layers[layerId]]
+  );
+  const currentIndexPatternId = state.layers[layerId].indexPatternId;
+
+  return (
+    <I18nProvider>
+      {isChooserOpen ? (
+        <EuiComboBox
```

Comment:
This combobox needs some extra styling, I think it's getting shorter as I type:
<img width="352" alt="Screenshot 2019-08-06 14 50 30" src="https://user-images.githubusercontent.com/666475/62567632-94d38100-b859-11e9-8070-b19c5ad394cf.png">